Anixe.Ion/IonWriter.cs

```diff
@@ -0,0 +1,276 @@
+            this.state |= WriterState.Property;
+        }
+
+        public void WriteProperty(string name, float value)
+        {
+            ValidateWriteProperty(name);
+            ClearState();
+            this.tw.Write(name);
+            this.tw.Write(Consts.IonSpecialChars.EqualsCharacter);
+            this.tw.WriteLine(value.ToString(CultureInfo.InvariantCulture));
+            this.state |= WriterState.Property;
+        }
+
+        public void WriteProperty(string name, char value)
+        {
+            ValidateWriteProperty(name);
+            ClearState();
+            this.tw.Write(name);
+            this.tw.Write(Consts.IonSpecialChars.EqualsCharacter);
+            this.tw.Write(Consts.IonSpecialChars.QuotationCharacter);
+            this.tw.Write(value);
+            this.tw.WriteLine(Consts.IonSpecialChars.QuotationCharacter);
+            this.state |= WriterState.Property;
+        }
+
+        public void WriteProperty(string name, decimal value)
+        {
+            ValidateWriteProperty(name);
+            ClearState();
+            this.tw.Write(name);
+            this.tw.Write(Consts.IonSpecialChars.EqualsCharacter);
+            this.tw.WriteLine(value.ToString(CultureInfo.InvariantCulture));
+            this.state |= WriterState.Property;
+        }
+
+        private void ValidateWriteProperty(string name)
+        {
+            if (this.state == WriterState.None)
+            {
+                throw new InvalidOperationException("Only section can be at the top of document");
+            }
+            if (string.IsNullOrEmpty(name))
+            {
+                throw new ArgumentNullException("Property name must be provided");
+            }
+        }
+
+        public void WriteTableHeader(string[] columns)
+        {
+            ValidateWriteTableHeader(columns);
+            ClearState();
+            tw.WriteLine();
+            WriteTableRow(columns, WriteCol);
+            WriteTableRow(columns, WriteHeaderSeparator, true);
+            this.state &= ~WriterState.Property;
+            this.state |= WriterState.TableHeader;
+            this.lastTableColumns = columns;
+        }
+
+        private void ValidateWriteTableHeader(string[] columns)
+        {
+            if (this.state == WriterState.None)
+            {
+                throw new InvalidOperationException("Only section can be at the top of document");
+            }
+            if (this.state.HasFlag(WriterState.TableHeader))
+            {
+                throw new InvalidOperationException("Table can have ony one header");
+            }
+            if (columns == null || columns.Length == 0)
+            {
+                throw new ArgumentNullException("Cannot create empty table header");
+            }
+        }
+
+        public void WriteTableRow(string[] data)
+        {
+            if (this.lastTableColumns == null && !this.state.HasFlag(WriterState.TableHeader))
```

Comment:
HasFlag(...) can be replaced with faster implementation regarding to this article http://grahammoore.com/improving-enum-hasflag/